oat-sa/tao-core-ui-fe
# Feature/tao 10454/implement search retention

**related to:** https://oat-sa.atlassian.net/browse/TAO-10454

This PR is a continuation of [this one](https://github.com/oat-sa/tao-core-ui-fe/pull/170), and includes:

- Implement search results preservation as detailed on ticket
- Refactor of `searchModal` component to match with other TAO FE components
- Refactor of tests to reach a 100% of coverage

This PR is a dependency of [this](https://github.com/oat-sa/tao-core/pull/2660) PR on `tao-core`, which makes this component accessible for TAO users.

**How to test**
To be able to test this PR locally you need to setup your TAO instance to manage the new Advanced search feature. To do so you need to:
- On your root `composer.json` set the following dependencies:
```
    "oat-sa/tao-core": "feature/TAO-10453/implement-query-builder",
    "oat-sa/generis": "dev-feature/TAO-10203-advanced-search as 12.33.2",
    "oat-sa/extension-tao-dac-simple": "dev-feature/TAO-10203-advanced-search",
    "oat-sa/lib-tao-elasticsearch": "dev-feature/TAO-10203-advanced-search as 1.1.1"
```
- run `composer install`
- if you work with `taoDockerize` run `taoDockerize install`  (database will be reset)
- If you do not user `taoDockerize` run `php tao/scripts/taoUpdate.php`

Now `searchModal` component should be used on search trigger instead of previous search component. Notice that advanced search syntax is not enabled if you do not set up an elastic search server, which is not required to test this feature.

Please let me know if you need more details about how to test the feature.

![search-retention](https://user-images.githubusercontent.com/14041944/91667796-6af59800-eb07-11ea-8fcd-ce2c3feccbe3.gif)
